src/device.rs

```diff
@@ -563,22 +563,25 @@ impl<B: UsbBus> UsbDevice<'_, B> {
 
                 // rest STRING Requests
                 _ => {
-                    let lang_id = match LangID::try_from(req.index) {
-                        Err(_err) => {
-                            #[cfg(feature = "defmt")]
-                            defmt::warn!(
-                                "Receive unknown LANGID {:#06X}, reject the request",
-                                _err.number
-                            );
-                            xfer.reject().ok();
-                            return;
-                        }
+                    let lang_id = LangID::try_from(req.index);
 
-                        Ok(req_lang_id) => req_lang_id,
-                    };
                     let string = match index {
                         // Manufacturer, product, and serial are handled directly here.
                         1..=3 => {
+                            let lang_id = match lang_id {
+                                Err(_err) => {
+                                    #[cfg(feature = "defmt")]
+                                    defmt::warn!(
+                                        "Receive unknown LANGID {:#06X}, reject the request",
+                                        _err.number
+                                    );
+                                    xfer.reject().ok();
+                                    return;
+                                }
+
+                                Ok(req_lang_id) => req_lang_id,
```

Comment:
Does this code not introduce the same problem that you're describing in the PR? I.e. if the lang_id is invalid for the string IDs 1..3, wouldn't this also reject the transfer unexpectedly?

Reply:
you are correct, it does reject for IDs 1, 2 and 3. I just limited the changes of #122 to those 3 requests since I thought  they were "standard" and the lang_id would be expected to be valid.

Reply:
I think that'd be best